File: combat/src/classicthunder/combat/cards/DefendCard.java
package classicthunder.combat.cards;

import classicthunder.combat.combat.character.CharacterStats;
import classicthunder.combat.combat.character.CharacterType;
import com.badlogic.gdx.graphics.Texture;

public class DefendCard extends Card {

    public DefendCard(Texture texture) {
        super(texture, 1);
    }

    @Override
    public int GetEnergyCost() {
        return 1;
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {
        return (stats.getCharacterType() == CharacterType.PLAYER);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustBlock(5);
    }
}

File: combat/src/classicthunder/components/card/DefendCard.kt
package classicthunder.components.card

import classicthunder.components.character.CharacterStats
import classicthunder.components.character.CharacterType
import com.badlogic.gdx.graphics.Texture

class DefendCard(texture: Texture) : Card(texture, 1) {
    override fun GetEnergyCost(): Int {
        return 1
    }

    override fun CanApplyEffects(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.PLAYER
    }

    override fun ApplyEffects(stats: CharacterStats) {
        stats.AdjustBlock(5)
    }
}